File: ROT13/Zig/rot13.zig
const std = @import("std");

pub const ROT13 = struct {
    input: []const u8,

    pub fn init(input: []const u8) ROT13 {
        return ROT13{ .input = input };
    }

    fn transform(input: []const u8, allocator: std.mem.Allocator) ![]u8 {
        var output = std.ArrayList(u8).init(allocator);
        defer output.deinit();

        for (input) |c| {
            if (c >= 'a' and c <= 'z') {
                const encoded = ((c - 'a' + 13) % 26) + 'a';
                try output.append(encoded);
            } else if (c >= 'A' and c <= 'Z') {
                const encoded = ((c - 'A' + 13) % 26) + 'A';
                try output.append(encoded);
            } else {
                try output.append(c);
            }
        }

        return output.toOwnedSlice();
    }

    pub fn encode(self: *const ROT13, allocator: std.mem.Allocator) ![]u8 {
        return try ROT13.transform(self.input, allocator);
    }

    pub fn decode(self: *const ROT13, encoded_version: []const u8, allocator: std.mem.Allocator) ![]u8 {
        _ = self; 
        return try ROT13.transform(encoded_version, allocator);
    }
};

File: ROT13/С/rot13.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

struct rot13
{
    size_t len{};
    char *input{};

    void init()
    {
        input = (char*) malloc(1000);
        if (!input)
        {
            exit(1);
        }

        fgets(input, 1000, stdin);
        input[strcspn(input, "\n")] = '\0';
        len = strlen(input);
    }

    char *encode() const
    {
        char *encoded = malloc(len + 1);

        for (size_t i = 0; i < len; i++)
        {
            if (input[i] >= 'a' && input[i] <= 'z')
            {
                encoded[i] = 'a' + (input[i] - 'a' + 13) % 26;
            }
            else if (input[i] >= 'A' && input[i] <= 'Z')
            {
                encoded[i] = 'A' + (input[i] - 'A' + 13) % 26;
            }
            else
            {
                encoded[i] = input[i];
            }
        }
        encoded[len] = '\0';
        return encoded;
    }

    char *decode(const char *encoded_version) const
    {
        char *decoded = malloc(len + 1);

        for (size_t i = 0; i < len; i++)
        {
            if (encoded_version[i] >= 'a' && encoded_version[i] <= 'z')
            {
                decoded[i] = 'a' + (encoded_version[i] - 'a' + 13) % 26;
            }
            else if (encoded_version[i] >= 'A' && encoded_version[i] <= 'Z')
            {
                decoded[i] = 'A' + (encoded_version[i] - 'A' + 13) % 26;
            }
            else
            {
                decoded[i] = encoded_version[i];
            }
        }
        decoded[len] = '\0';
        return decoded;
    }
};
